src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -156,32 +158,49 @@ private async Task StartHostAsync(CancellationToken cancellationToken, int attem
             }
             catch (OperationCanceledException)
             {
+                GetHostLogger(localHost).LogDebug("Host startup was canceled.");
                 throw;
             }
             catch (Exception exc)
             {
-                LastError = exc;
-                State = ScriptHostState.Error;
-                attemptCount++;
+                bool isActiveHost = localHost == _host;
```

Comment:
For safety/clarity, user ReferenceEquals

Reply:
👍